### PR TITLE
[WIP] Elixir build

### PR DESCRIPTION
**Pull request by sivsushruth**
```
mix deps.get
mix release
rel/barrel/bin/barrel console
```

~~TODO: production env profile~~
~~BUG: Logs not showing up in console, but visible on the admin dashboard logs page~~

### CODE REVIEW

**Comment by benoitc** [comment]
logs are disable in console. You need to enable them in the configuration file.

**Comment by sivsushruth** [comment]
So we have 
```
mix rel # for prod release
mix devrel #for dev release
```

The default exrm mix release.clean etc work too.

**Comment by sivsushruth** [comment]
We now have 

```
mix deps.get
mix compile
mix rel
```

```
mix deps.get
MIX_ENV=prod mix compile
MIX_ENV=prod mix rel
```

Thoughts ?

**Comment by benoitc** [comment]
what's the default env?

**Comment by sivsushruth** [comment]
@benoitc  the default env is dev

File: config/relx.config [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+           %% keep empty files
+           {copy, "../../config/empty", "log/KEEP"},
+           {copy, "../../config/empty", "data/KEEP"},
+
+           %% config files
+           {template, "../../config/barrel.ini", "etc/barrel.ini"},
+           {template, "../../config/local.ini", "etc/local.ini"},
+            {template, "../../config/vm.args", "etc/vm.args"},
+            {template, "../../config/sys.config", "etc/sys.config"}
```
Comment (benoitc) [inline config/relx.config]: why relative paths? What about having it in the root instead?

File: mix.exs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        :exometer_core
+      ]
+    ]
+  end
+
+  defp deps do
+    [
+      {:lager, "3.0.2"},
+      {:exrm, "~> 0.18.1"},
+      {:barrel_nifs, path: "utilities/barrel_release_plugin"},
```
Comment (benoitc) [inline mix.exs]: should be in support

File: utilities/barrel_release_plugin/mix.exs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  end
+
+  # Configuration for the OTP application
+  #
+  # Type "mix help compile.app" for more information
+  def application do
+    [applications: [:logger]]
+  end
+
+  # Dependencies can be Hex packages:
```
Comment (benoitc) [inline utilities/barrel_release_plugin/mix.exs]: remove the comment please :)

**Comment by benoitc** [comment]
I get the following error:

```
[barrel-platform] mix dep.get                                                                             18:26:19  ☁  elixir_build ☂ 𝝙 ✖ ➜ ✚
** (SyntaxError) mix.exs:85: unexpected token: "]". "do" starting at line 1 is missing terminator "end"
    (elixir) lib/code.ex:321: Code.load_file/2
    (mix) lib/mix/cli.ex:26: Mix.CLI.proceed/1
    (elixir) lib/code.ex:363: Code.require_file/2
```

any idea why?

**Comment by benoitc** [comment]
nvm, the branch was broken there.

**Pull request merged**